apollo-router/src/plugins/rhai.rs

```diff
@@ -387,21 +426,62 @@ impl Plugin for Rhai {
         };
 
         let main = scripts_path.join(&main_file);
-        let sdl = init.supergraph_sdl.clone();
-        let engine = Arc::new(Rhai::new_rhai_engine(Some(scripts_path)));
-        let ast = engine.compile_file(main)?;
-        let mut scope = Scope::new();
-        scope.push_constant("apollo_sdl", sdl.to_string());
-        scope.push_constant("apollo_start", Instant::now());
 
-        // Run the AST with our scope to put any global variables
-        // defined in scripts into scope.
-        engine.run_ast_with_scope(&mut scope, &ast)?;
+        let watched_path = scripts_path.clone();
+        let watched_main = main.clone();
+        let watched_sdl = sdl.clone();
+
+        let block = Arc::new(ArcSwap::from_pointee(EngineBlock::try_new(
+            Some(scripts_path),
+            main,
+            sdl,
+        )?));
+        let watched_block = block.clone();
+
+        let park_flag = Arc::new(AtomicBool::new(false));
+        let watching_flag = park_flag.clone();
+
+        let watcher_handle = std::thread::spawn(move || {
```

Comment:
I think it would be better to use something more async compliant like this example https://github.com/notify-rs/notify/blob/main/examples/async_monitor.rs and use https://docs.rs/tokio/0.2.22/tokio/task/fn.block_in_place.html instead of `block_on` in the example. (I'm not sure it needs to be checked). But by using this async compliant solution you won't have to start your own thread and park it by yourself.